src/parser/ad_parser.js

```diff
@@ -237,47 +236,70 @@ function parseWrapper(wrapperElement) {
  */
 function parseExtensions(collection, extensions) {
```

Comment:
I don't think we should pass collection to this function. we should just return extension, and in the place we call `parseExtensions` we check if the returned value is truthy, then push it to `ad.extensions`. This is because it isn't clear that we are mutating `ad.extensions` by calling this function